### PR TITLE
External url

### LINKED ISSUE #26: Enable access via `kubectl proxy`
When the kubes cluster isn't running locally, it's pretty common to need to access it via the apiserver's proxy mechanism - this is particularly useful for "infrastructure" tools, as the authentication mechanism of Kubernetes is applied.

However, the Prometheus UI uses absolute URLs with what I would go so far as to call reckless abandon, and doesn't work at all behind any proxy which doesn't rewrite all requests.

It can be made to work by adding these two arguments to prometheus:

    -web.external-url=http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/prometheus-k8s:web/
    -web.route-prefix=/

And then the UI can be reached by running `kubectl proxy` and browsing to the URL above.

### PR DESCRIPTION
**Pull request by brancz**
@alexsomesan @fabxc 

Tested with minikube. And the following manifests (replacing the manifests from `kube-prometheus`).

Prometheus:

```
apiVersion: monitoring.coreos.com/v1alpha1
kind: Prometheus
metadata:
  name: prometheus-k8s
  labels:
    prometheus: k8s
spec:
  version: v1.4.1
  externalUrl: http://192.168.99.100:30900/prometheus
  resources:
    requests:
      memory: 400Mi
```

Alertmanager:

```
apiVersion: "monitoring.coreos.com/v1alpha1"
kind: "Alertmanager"
metadata:
  name: "alertmanager-main"
  labels:
    alertmanager: "main"
spec:
  replicas: 3
  version: v0.5.1
  externalUrl: http://192.168.99.100:30903/alertmanager
```

Fixes #26

### CODE REVIEW

File: pkg/alertmanager/statefulset.go [hunk 1]
```diff
@@ -117,6 +117,11 @@ func makeStatefulSetSpec(a *spec.Alertmanager) v1beta1.StatefulSetSpec {
 		fmt.Sprintf("-mesh.listen-address=:%d", 6783),
 		fmt.Sprintf("-storage.path=%s", "/etc/alertmanager/data"),
 	}
+
+	if a.Spec.ExternalUrl != "" {
```
Comment (fabxc) [inline pkg/alertmanager/statefulset.go]: `ExternalURL`.

File: pkg/alertmanager/statefulset.go [hunk 2]
```diff
@@ -117,6 +117,11 @@ func makeStatefulSetSpec(a *spec.Alertmanager) v1beta1.StatefulSetSpec {
...
+		commands = append(commands, "-web.external-url="+a.Spec.ExternalUrl)
```
Comment (fabxc) [inline pkg/alertmanager/statefulset.go]: That should probably come in quotes to be sure.

File: pkg/prometheus/statefulset.go [hunk 3]
```diff
@@ -196,7 +207,7 @@ func makeStatefulSetSpec(p spec.Prometheus) v1beta1.StatefulSetSpec {
 						ReadinessProbe: &v1.Probe{
 							Handler: v1.Handler{
 								HTTPGet: &v1.HTTPGetAction{
-									Path: "/status",
+									Path: webRoutePrefix + "/status",
```
Comment (fabxc) [inline pkg/prometheus/statefulset.go]: Wrapped into `path.Clean`

File: pkg/alertmanager/statefulset.go (hunk 2 above)
Comment (brancz) [inline pkg/alertmanager/statefulset.go]: This is a sequence of commands as they are represented after the arguments have been parsed initially, so quoting results in the quotes being an actual part of the flags string value:

When trying this in the Prometheus StatefulSet:

```
level=error msg="invalid Prometheus URL: \"http://192.168.99.100:30900/prometheus\""
```

When trying this in the Alertmanager StatefulSet:

```
panic: wildcards must be named with a non-empty name in path '/"http://192.168.99.100:30903/alertmanager"/app/*filepath'
```

**Comment by brancz** [comment]
Re-review @fabxc ?

**Comment by fabxc** [comment]
👍

**Pull request merged**